controllers/comments-controller.js

```diff
@@ -0,0 +1,39 @@
+const {
+  fetchCommentsByArticleId,
+  fetcharticlesIds,
+} = require("../models/comments-model.js");
+
+const getCommentsByArticleId = (req, res, next) => {
+  const { articleId } = req.params;
+  const promiseFilteredComments = fetchCommentsByArticleId(articleId);
+  const promiseArticlesIds = fetcharticlesIds();
+  Promise.all([promiseFilteredComments, promiseArticlesIds])
+    .then((result) => {
+      const [result1, result2] = result;
+      const { rows: filteredComments } = result1;
+      const { rows: articlesIds } = result2;
+      const idInIdsArray = articlesIds.some(
+        (article) => article.article_id === Number(articleId)
+      );
+
+      if (filteredComments.length === 0 && !idInIdsArray) {
```

Comment:
I would suggest checking this in the models rather than controllers

Reply:
Thanks!